### ISSUE TITLE
No support for named captures

### ISSUE THREAD

**Issue by akimd** [ruby/strscan#43]
AFAICT, named capture groups can be defined in the RE, but cannot be recovered from the strscan.  Is this only an oversight, or something decided?

Cheers!

**Comment by kou** [comment]
Could you provide a script that shows this case?

**Comment by akimd** [comment]
Hi!

``` ruby
require 'strscan'

s = 'foobarbaz'
re = /(?<f>foo)(?<r>bar)(?<z>baz)/

m = re.match s
p m.captures
# ["foo", "bar", "baz"]
p m.named_captures
# {"f"=>"foo", "r"=>"bar", "z"=>"baz"}

scan = StringScanner.new(s)
scan.match? re
p scan.captures
# ["foo", "bar", "baz"]
p scan.named_captures
# /tmp/foo.rb:13:in `<main>': undefined method `named_captures' for #<StringScanner 0/9 @ "fooba..."> (NoMethodError)
# 
# p scan.named_captures
#       ^^^^^^^^^^^^^^^
```

**Comment by kou** [comment]
Thanks!
We should support this case.

**Comment by kou** [comment]
@headius Do you or anyone who is interested in JRuby want to implement this feature? You can enable this feature's test for JRuby by changing https://github.com/ruby/strscan/commit/1c69ee2a9a2f9b3d3382f872f003d42412c62a3e#diff-1106672f9e23f0054b7ff95f28e63ebea6cb046526017634aa8b608e37748086R788 .

**Comment by headius** [comment]
@kou I can look into it.

**Comment by headius** [comment]
@kou Implemented in b217522. Thanks for the ping!

**Comment by kou** [comment]
Thanks!
